### PR TITLE
develop-1.0 tweaks

### PR DESCRIPTION
**Pull request by vineetbansal**
These changes bring in some of the current `master` branch tweaks to this branch, so that later merges back to `master` are less painful. Most of the changes are cosmetic (whitespace etc.), addition/deletion of CI files, etc. The tests on this branch still pass, and can be verified.

### CODE REVIEW

File: docs/citing/index.rst [hunk 1]
```diff
@@ -57,3 +71,17 @@ If you use OSQP for published work, we encourage you to put a star on `GitHub <h
             doi       = {10.1109/CDC.2017.8263928},
             url       = {https://doi.org/10.1109/CDC.2017.8263928},
           }
+
+    Mixed-integer optimization
+        A branch-and-bound solver for mixed-integer quadratic optimization in this `paper <https://stellato.io/assets/downloads/publications/2018/miosqp_ecc.pdf>`__.
```
Comment (gbanjac) [inline docs/citing/index.rst:76]: @bstellato Can you please fix this link? Also related to https://github.com/osqp/osqp/issues/329

File: docs/interfaces/CC++.rst [hunk 2]
```diff
@@ -1,6 +1,6 @@
-.. _c_cpp_interface:
+.. _c_interface:
 
-C/C++
+C
```
Comment (gbanjac) [inline docs/interfaces/CC++.rst:3]: How about we also change the name of the file to `C.rst`?

File: docs/interfaces/CC++.rst [hunk 3]
```diff
@@ -25,14 +25,18 @@ The main C/C++ API is imported from the header :code:`osqp.h` and provides the f
...
 Warm start
 ^^^^^^^^^^
 OSQP automatically warm starts primal and dual variables from the previous QP solution. If you would like to warm start their values manually, you can use
 
 .. doxygenfunction:: osqp_warm_start
 
+.. doxygenfunction:: osqp_warm_start_x
+
+.. doxygenfunction:: osqp_warm_start_y
+
```
Comment (gbanjac) [inline docs/interfaces/CC++.rst:39]: I thought we decided to remove these function here: https://github.com/osqp/osqp/commit/fab90f8b138aa48c9be71c1d1ddbbe45267a1308

File: docs/interfaces/CC++.rst [hunk 4]
```diff
@@ -42,6 +46,10 @@ Problem data can be updated without executing the setup again using the followin
 
 .. doxygenfunction:: osqp_update_lin_cost
 
+.. doxygenfunction:: osqp_update_lower_bound
+
+.. doxygenfunction:: osqp_update_upper_bound
+
```
Comment (gbanjac) [inline docs/interfaces/CC++.rst:52]: Similar comment as above. I think we can just keep `osqp_update_bounds` and set the corresponding arguments to `NULL` if we only want to update one bound.

File: docs/interfaces/CC++.rst (hunk 3 above)
Comment (gbanjac) [reply docs/interfaces/CC++.rst:39]: You can see that these function do not appear in `osqp_api_functions.h` any more.

File: docs/citing/index.rst (hunk 1 above)
Comment (gbanjac) [reply docs/citing/index.rst:76]: The new link should be `https://stellato.io/downloads/publications/2018/miosqp_ecc.pdf`.

Comment (vineetbansal) [reply docs/citing/index.rst]: Done here - will send separate PR for master.

File: docs/interfaces/CC++.rst (hunk 2 above)
Comment (vineetbansal) [reply docs/interfaces/CC++.rst:3]: Done here - will send separate PR for master.

File: docs/interfaces/CC++.rst (hunk 3 above)
Comment (vineetbansal) [reply docs/interfaces/CC++.rst]: Done. Not changing this in master just yet.

File: docs/interfaces/CC++.rst (hunk 4 above)
Comment (vineetbansal) [reply docs/interfaces/CC++.rst]: Done. Not changing this in master just yet.

File: docs/citing/index.rst (hunk 1 above)
Comment (gbanjac) [reply docs/citing/index.rst]: Please update the link here as well:
https://github.com/osqp/osqp/blob/master/site/_pages/citing.md

Comment (vineetbansal) [reply docs/citing/index.rst]: Fixed in this branch. Will fix in master separately.

**Comment by vineetbansal** [comment]
@goulart-paul - 2 tests are failing on MacOS? (for EMBEDDED=1/2):
```
/Users/runner/work/osqp/osqp/algebra/default/matrix.c:11:11: error: implicit declaration of function 'csc_is_eq' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
          csc_is_eq(A->csc, B->csc, tol) );
          ^
```          
Since `csc_is_eq` is only defined for non-embedded cases, does something needs to be done here?

**Comment by vineetbansal** [comment]
Thanks @gbanjac - now that the tests are passing, should we merge this now? I don't want to have my branch of `develop-1.0` hanging around. You can continue making commits directly to `develop-1.0` of course, and I'll start sending PRs for groups of cherry-picked commits from `master` soon. I'll try to minimize the no. of PRs I create.

**Comment by gbanjac** [comment]
Sounds good. I will merge it now.

**Pull request merged**